### DISCUSSION TITLE
Redeploy a contract

### DISCUSSION THREAD

**Discussion by manojkgorle** [NomicFoundation/hardhat#4387] [General]
Hey I want to redeploy few contracts, when I am doing npx hardhat deploy --network <network>
It is not deploying any new contract but showing address of previous contracts

I tried deleting the deployments of the network, did not work
I ran npx hardhat --clean , did not work, 

what can i do?

**Comment by fvictorio** [comment]
You need to start a node in one terminal with `npx hardhat node` and then run your deployments with `--network localhost` so that the node is used. Otherwise each deployment is done in an ephemeral hardhat network which is destroyed when the task finishes.
